### ISSUE TITLE
Question: compatibility with org.apache.commons:commons-imaging ?

### ISSUE THREAD

**Issue by KoenDG** [haraldk/TwelveMonkeys#823]
Not an issue, more a question:

To be specific, this library: https://mvnrepository.com/artifact/org.apache.commons/commons-imaging

Homepage: https://commons.apache.org/proper/commons-imaging/

Should a Java project import that and then also any of the plugins here, would that cause any issues? Or is it a requirement?

**Comment by haraldk** [comment]
Hi Koen,

TwelveMonkeys ImageIO and Commons Imaging provides similar functionality, in that both provides read/write of images and metadata. They both support many of the same formats, and decodes/encodes to/from `java.awt.image.BufferedImage`.

However, there is an important difference: TwelveMonkeys ImageIO provides plugins for a standard API (the `javax.imageio` API). It can be extended by and is Interchangeable with other implementations. You can choose the plugins for the formats you need in your application.

While Commons Imaging provides a completely alternative API, and is limited to their own implementation. 

TwelveMonkeys ImageIO have no dependencies on Commons Imaging. And there are not other dependencies between them that I’m aware of. It should be possible to use both in the same project, but I'm not sure if it makes sense, as the functionality overlaps, and client code would have to interface with two different APIs. 

So to sum up: **Should you? No. Would it cause any issue? Probably not, aside from more complicated code. Is it a requirement. Absolutely not.** Based on your requirements, I would recommend you choose the one that supports the formats you need and pick the API you prefer to work with (I know what I would choose, but I might be a little biased in this question... 😉 ).

**Comment by KoenDG** [comment]
Thanks for the answer, it's good to be sure of things.

Looking at their code examples, it does seem one needs to import `org.apache.commons.imaging.*` an access their API to work with it.

So, a separate api, as you say.
